plugins/pagerduty/api-report.md

```diff
@@ -64,6 +64,8 @@ export class PagerDutyClient implements PagerDutyApi {
   //
   // (undocumented)
   getServiceByIntegrationKey(integrationKey: string): Promise<Service[]>;
+  // (undocumented)
+  getServiceByServiceId(serviceId: string): Promise<Service>;
```

Comment:
@Rugvip thanks for looking!

>On a separate note, return type here is a bit suspicious, the old one returns an array, but this one is just a single service?

Yeah, the original implementation queries the [List Services Endpoint](https://developer.pagerduty.com/api-reference/e960cca205c0f-list-services) with a filter on the integration key, so the return value of that API is an array of services.

This new implementation is specific in querying an exact service using the [Get Service Endpoint](https://developer.pagerduty.com/api-reference/165ad96a22ffd-get-a-service), so there is only one object to return.

>I feel it would be better to move this all the way to getServiceByEntity()

Noted! I didn't feel comfortable making that decision on my own, so I just went ahead an bolted on an option to better support existing behavior.

I'll go ahead and refactor a bit, but I think that its currently configured this way to isolate dependencies on React and the API Client.

> essentially baking the logic from `usePagerdutyEntity` into the API implementation rather than having that live in React land

This may not be _totally_ feasible since [`usePagerdutyEntity` relies on useEntity from the `@backstage/plugin-catalog-react`](https://github.com/backstage/backstage/blob/32bff53079c7b6890067f515be06e291804faed2/plugins/pagerduty/src/hooks/index.ts#L17), so there is an implied reliance on doing some of this login _in react_. 

What you're suggesting feels like a whole refactor of the API client, since there are other API calls that rely on on providing a `serviceID` to query data (e.g. [`getIncidentsByServiceId`](https://github.com/backstage/backstage/blob/79b0e20cb2bb62963e223b884dd6a542f2119f42/plugins/pagerduty/src/api/client.ts#L83) and [`getChangeEventsBySeviceId`](https://github.com/backstage/backstage/blob/79b0e20cb2bb62963e223b884dd6a542f2119f42/plugins/pagerduty/src/api/client.ts#L93))

I'm inclined to take what you're suggesting, but moreso have the `getServiceByEntity` take an object of type `PagerDutyEntity` that we can normalize.

I'll take a stab at putting something together!